template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/general/GeneralClusterConfigsProvider.java

```diff
@@ -53,6 +54,9 @@ public GeneralClusterConfigs generalClusterConfigs(Stack stack, Cluster cluster)
         generalClusterConfigs.setKnoxUserFacingCertConfigured(userFacingCertHasBeenGenerated);
         generalClusterConfigs.setExternalFQDN(cluster.getFqdn());
         generalClusterConfigs.setEnableRangerRaz(cluster.isRangerRazEnabled());
+        generalClusterConfigs.setUserCrn(stack.getEnvironmentCrn());
```

Comment:
I don't think it will work with the environment crn. The stack has a creator crn instead, but I still recommend to use the `com.sequenceiq.cloudbreak.auth.ThreadBasedUserCrnProvider.INTERNAL_ACTOR_CRN` here.

Reply:
Now, I am using INTERNAL_ACTOR_CRN as crn and ThreadBasedUserCrnProvider.getAccountId() as account ID. This is how entitlement is checked at
https://github.com/hortonworks/cloudbreak/blob/master/core/src/main/java/com/sequenceiq/cloudbreak/converter/v4/stacks/instancegroup/template/InstanceTemplateV4RequestToTemplateConverter.java#L66

Reply:
Hmm, nice trick! I was not aware of `ThreadBasedUserCrnProvider.getAccountId()`, but it is definitely smarter this way. 😃

Reply:
unless there is an issue and it contains the wrong account id :D